hpack/table.py

```diff
@@ -0,0 +1,207 @@
+        (b'date'                        , b''             ),
+        (b'etag'                        , b''             ),
+        (b'expect'                      , b''             ),
+        (b'expires'                     , b''             ),
+        (b'from'                        , b''             ),
+        (b'host'                        , b''             ),
+        (b'if-match'                    , b''             ),
+        (b'if-modified-since'           , b''             ),
+        (b'if-none-match'               , b''             ),
+        (b'if-range'                    , b''             ),
+        (b'if-unmodified-since'         , b''             ),
+        (b'last-modified'               , b''             ),
+        (b'link'                        , b''             ),
+        (b'location'                    , b''             ),
+        (b'max-forwards'                , b''             ),
+        (b'proxy-authenticate'          , b''             ),
+        (b'proxy-authorization'         , b''             ),
+        (b'range'                       , b''             ),
+        (b'referer'                     , b''             ),
+        (b'refresh'                     , b''             ),
+        (b'retry-after'                 , b''             ),
+        (b'server'                      , b''             ),
+        (b'set-cookie'                  , b''             ),
+        (b'strict-transport-security'   , b''             ),
+        (b'transfer-encoding'           , b''             ),
+        (b'user-agent'                  , b''             ),
+        (b'vary'                        , b''             ),
+        (b'via'                         , b''             ),
+        (b'www-authenticate'            , b''             ),
+    )
+
+    def __init__(self):
+        self._maxsize = HeaderTable.DEFAULT_SIZE
+        self.resized = False
+        self.dynamic_entries = deque()
+
+    def get_by_index(self, index):
+        """
+        Returns the entry specified by index
+
+        Note that the table is 1-based ie an index of 0 is
+        invalid.  This is due to the fact that a zero value
+        index signals that a completely unindexed header
+        follows.
+
+        The entry will either be from the static table or
+        the dynamic table depending on the value of index.
+        """
+        index -= 1
+        if index < 0:
+            return None # TODO throw HPACKException here
+        if index < len(HeaderTable.STATIC_TABLE):
+            return HeaderTable.STATIC_TABLE[index]
+        index -= len(HeaderTable.STATIC_TABLE)
+        if index < len(self.dynamic_entries):
+            return self.dynamic_entries[index]
+        return None # TODO throw HPACKException here
+
+    def __repr__(self):
+        return "HeaderTable(%d, %s, [%s])" % (
+            self._maxsize,
+            self.resized,
+            ", ".join([str(e) for e in self.dynamic_entries])
```

Comment:
I think you can skip the overhead of doing the `join`, just change the third `%s` to a `%r` and Python will do that for you.